libs/core/uagents_core/utils/communication.py

```diff
@@ -0,0 +1,87 @@
+import json
+from typing import Optional, Any
+from uuid import uuid4, UUID
+
+import requests
+import urllib.parse
+
+from uagents_core.crypto import Identity
+from uagents_core.envelope import Envelope
+from uagents_core.config import DEFAULT_AGENTVERSE_URL, DEFAULT_ALMANAC_API_PATH
+from uagents_core.logger import get_logger
+
+
+
+logger = get_logger("uagents_core.utils.communication")
+
+
+def lookup_endpoint_for_agent(agent_address: str, *, agentverse_url: Optional[str] = None) -> str:
+    agentverse_url = agentverse_url or DEFAULT_AGENTVERSE_URL
+    almanac_api = urllib.parse.urljoin(agentverse_url, DEFAULT_ALMANAC_API_PATH)
+
+    request_meta = {
+        "agent_address": agent_address,
+        "lookup_url": almanac_api,
+    }
+    logger.debug("looking up endpoint for agent", extra=request_meta)
+    r = requests.get(f"{almanac_api}/agents/{agent_address}")
+    r.raise_for_status()
+
+    request_meta["response_status"] = r.status_code
+    logger.info(
+        "Got response looking up agent endpoint",
+        extra=request_meta,
+    )
+
+    return r.json()["endpoints"][0]["url"]
+
+
+def send_message_to_agent(
+    sender: Identity,
+    target: str,
+    payload: Any,
+    protocol_digest: str,
+    model_digest: str,
+    session: UUID = uuid4(),
+    *,
+    agentverse_url: Optional[str] = None,
+):
```

Comment:
I don't mind having a duplicate sync version of this function, but I'd suggest we use the same or at least very similar naming conventions and API as the async version: https://github.com/fetchai/uAgents/blob/main/python/src/uagents/communication.py#L267-L275